sdk/core/core/CMakeLists.txt

```diff
@@ -62,7 +62,7 @@ else()
   target_compile_options(az_core PRIVATE -Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG})
 endif()
 
-target_include_directories (az_core PUBLIC inc)
+target_include_directories (az_core PUBLIC inc inc/internal)
```

Comment:
This is a problem if you want to actually hide stuff

a customer could do `#include <az_span_def.h>` just fine.

I'm not a huge fan of trying to hide stuff like this in the first place but if you must then this could either be `PRIVATE`, or even better just keep the include directories as just `inc` and have public headers include stuff with `#include "internal/..."`